### ISSUE TITLE
Minor nitpick about FMA doc / comment

### ISSUE THREAD

**Issue by lycium** [madmann91/bvh#91]
https://github.com/madmann91/bvh/blob/ce037a75d20ce23e404103bcd64f8fef646aac04/src/bvh/v2/utils.h#L65

The F in FMA stands for "fused", which refers to the more important part of the operation than its performance (i.e. not "fast"), namely that the result of the multiplication and the addition are fused together into a single operation, with all intermediate bits and no rounding inbetween.

P.S. Love your library, some of the most beautiful code I've ever seen <3

**Comment by madmann91** [comment]
Hello, and thank you for your comment.

The name `fast_multiply_add` here just reflects that the implementation picks whichever is fastest: A fused multiply-add (FMA), or a mul followed by an addition. Note that `FP_FAST_FMAF` should be defined if FMA is faster than a multiplication followed by an addition (left as implementation-defined according to the standard). Some compilers do not define `FP_FAST_FMAF`, which is why this code also uses `#pragma` directives to allow the compiler to contract the regular addition and multiplication into a single FMA _if it decides this is going to be faster_ (and those `#pragma` directives sadly differ from compiler to compiler).

To sum things up, this code does not always generate an FMA, and perhaps the comment should reflect that. If my attempt with this code had been to use FMA always, I could have just used `std::fma(f)` directly. However, it would have been emulated on instruction sets that do not support such an instruction natively.
